HW05.Rmd

```diff
@@ -0,0 +1,145 @@
+---
+title: "HW05: mom_dad and CTRP data"
+author: "Laura N. Cespedes Arias"
+date: "31/7/2020"
+output: github_document
+---
+
+```{r setup, include=TRUE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Loading packages and importing data
+```{r packages, message=FALSE, warning=FALSE}
+library(tidyverse)    
+library(knitr)
+```
+
+# 1) The mom_dad data 
+
+## Importing the data
+```{r include=TRUE}
+# read in the data
+dad_mom <- read.table("dad_mom.txt",h=T)
+
+#These is how the table looked originally:
+knitr::kable(dad_mom)
+```
+
+## Tidying the dad_mom file
+
+Pretty sure there is a shorter and cleaner way to do this, but this is was the best solution I could think of:
+
+First, I used pivot_longer to reduce the number of columns (now each mom and dad per family have their own row). I created a column named "parent" where I specified if the person was a mom or a dad. Then I replaced the values of these variable (using mutate & replace). After that, I created a new column called "income" which has the value for the income of the corresponding parent (extracting values from income_dad and income_mom depending of the "parent" column) For this I used mutate & if_else. Lastly, I eliminated the income_dad and income_mom columns.
+
+```{r include=TRUE}
+ dad_mom_v2 <-
+  dad_mom %>% pivot_longer(c(name_dad, name_mom),
+                           names_to = "parent",
+                           values_to = "name") %>% mutate(parent = replace(parent, parent == "name_mom", "mom")) %>% mutate(parent = replace(parent, parent == "name_dad", "dad")) %>% mutate(income = if_else(parent == "mom", income_mom, income_dad)) %>% select(-c(income_dad, income_mom))
```

Comment:
Although this is a bit longer than the way I went about it, I love moments like these because they really show the flexibility of R